cloud.foundry.cli/src/test/java/cloud/foundry/cli/system/util/SpaceManager.java

```diff
@@ -0,0 +1,321 @@
+     * @param desiredServiceBean the bean of the desired service
+     * @return random name used for the service
+     */
+    public String requestCreationOfService(ServiceBean desiredServiceBean) {
+        String randomName = "service-" + makeRandomString();
+        servicesToCreate.put(randomName, desiredServiceBean);
+        return randomName;
+    }
+
+    /**
+     * Registers an application that is desired be created on the space.
+     * @param desiredApplicationBean the bean of the desired application
+     * @return random name used for the application
+     */
+    public String requestCreationOfApplication(ApplicationBean desiredApplicationBean) {
+        String randomName = "app-" + makeRandomString();
+        applicationsToCreate.put(randomName, desiredApplicationBean);
+        return randomName;
+    }
+
+    /**
+     * Creates all previously registered desired services and applications on the space. After the creation process has
+     * finished, the applications and services are not registered as desired anymore.
+     * @throws RuntimeException or other subclasses of RuntimeException if any errors occur during the creation process
+     */
+    public void createRequestedEntities() {
+        // FIXME if possible: Flux.merge would be faster but it leads to internal server errors on the cf instance
+        Flux.concat(collectServiceCreationRequests()).blockLast();
+
+        Flux.merge(collectApplicationCreationRequests()).blockLast();
+    }
+
+    private List<Mono<Void>> collectServiceCreationRequests() {
+        if (servicesOperations == null) {
+            throw new IllegalStateException("space has not been created yet");
+        }
+
+        List<Mono<Void>> resultingCreationRequests = servicesToCreate.entrySet().stream()
+                .map(serviceEntry -> servicesOperations.create(serviceEntry.getKey(), serviceEntry.getValue()))
+                .collect(Collectors.toList());
+
+        return resultingCreationRequests;
+    }
+
+    private List<Mono<Void>> collectApplicationCreationRequests() {
+        if (applicationsOperations == null) {
+            throw new IllegalStateException("space has not been created yet");
+        }
+
+        List<Mono<Void>> resultingCreationRequests = applicationsToCreate.entrySet().stream()
+                .map(applicationEntry ->
+                        applicationsOperations.create(applicationEntry.getKey(), applicationEntry.getValue(), false))
+                .collect(Collectors.toList());
+        return resultingCreationRequests;
+    }
+
+    /**
+     * Removes all previously registered desired services and applications on the space. After the removal process has
+     * finished, the applications and services are not registered as desired anymore.
+     * @throws RuntimeException or other subclasses of RuntimeException if any errors occur during the removal process
+     */
+    public void removeSpace() {
+        cleanUpSpace();
```

Comment:
Can the space be deleted without cleaning it up? Have you tried that?

Reply:
It cannot. I have tried that, yes. I already had removed this `clearAll` helper method, then had to bring it back for that reason. It even got extended.